packages/sourcecred/src/core/ledger/grainAllocation.test.js

```diff
@@ -61,35 +53,20 @@ describe("core/ledger/grainAllocation", () => {
     describe("validation", () => {
       beforeEach(async (done) => {
         credGraph = await GraphUtil.credGraph();
+        credGraph2 = await GraphUtil.credGraph2();
+
         credGrainView = CredGrainView.fromCredGraphAndLedger(
           credGraph,
           emptyLedger
         );
+
+        credGrainView2 = CredGrainView.fromCredGraphAndLedger(
+          credGraph2,
+          emptyLedger2
+        );
         done();
       });
 
-      it("errors if there are no identities", () => {
-        const thunk = () =>
-          computeAllocation(immediate(5), [], credGrainView, 0);
-        expect(thunk).toThrowError("must have at least one identity");
-      });
-      it("errors if the total cred is zero", () => {
-        const thunk = () =>
-          computeAllocation(immediate(5), [aid(0, [0])], credGrainView, 0);
-        expect(thunk).toThrowError("cred is zero");
-      });
-      it("errors if there's NaN or Infinity in Cred", () => {
-        const thunk = () =>
-          computeAllocation(immediate(5), [aid(0, [NaN])], credGrainView, 0);
-        expect(thunk).toThrowError("invalid cred");
-      });
-      it("errors if there's inconsistent Cred lengths", () => {
-        const i1 = aid(0, [1]);
-        const i2 = aid(0, [1, 2]);
```

Comment:
So I removed these checks, and added a validation routine in the CredGrainView to do some of the same checks, however those checks are not tested. To test some of we would have to create a a credGrainView with no cred, or with mismatched interval data for participants, and I'm not sure how to go about that. Thoughts?

Reply:
You can use the CredGrainView.fromJSON function to construct it manually with desired test data.